src/contracts/Protocol/QuestionStateController.sol

```diff
@@ -13,7 +13,7 @@ contract QuestionStateController is IQuestionStateController, Ownable, OnlyApi {
     mapping(address => mapping(uint256 => bool)) public hasVoted;
     mapping(address => mapping(uint256 => uint256)) public questionIndex;
 
-    mapping(STATE => QuestionData[]) public questionByState;
+    mapping(STATE => mapping(uint256 => QuestionData)) public questionByState;
```

Comment:
add additional mapping of questionID which we can get latest by calling getter on question API 
`await questionAPI.currentQuestionId();`